pkg/apis/pipeline/v1/taskrun_types.go

```diff
@@ -0,0 +1,473 @@
+)
+
+func (t TaskRunReason) String() string {
+	return string(t)
+}
+
+// GetStartedReason returns the reason set to the "Succeeded" condition when
+// InitializeConditions is invoked
+func (trs *TaskRunStatus) GetStartedReason() string {
+	return TaskRunReasonStarted.String()
+}
+
+// GetRunningReason returns the reason set to the "Succeeded" condition when
+// the TaskRun starts running. This is used indicate that the resource
+// could be validated is starting to perform its job.
+func (trs *TaskRunStatus) GetRunningReason() string {
+	return TaskRunReasonRunning.String()
+}
+
+// MarkResourceOngoing sets the ConditionSucceeded condition to ConditionUnknown
+// with the reason and message.
+func (trs *TaskRunStatus) MarkResourceOngoing(reason TaskRunReason, message string) {
+	taskRunCondSet.Manage(trs).SetCondition(apis.Condition{
+		Type:    apis.ConditionSucceeded,
+		Status:  corev1.ConditionUnknown,
+		Reason:  reason.String(),
+		Message: message,
+	})
+}
+
+// MarkResourceFailed sets the ConditionSucceeded condition to ConditionFalse
+// based on an error that occurred and a reason
+func (trs *TaskRunStatus) MarkResourceFailed(reason TaskRunReason, err error) {
+	taskRunCondSet.Manage(trs).SetCondition(apis.Condition{
+		Type:    apis.ConditionSucceeded,
+		Status:  corev1.ConditionFalse,
+		Reason:  reason.String(),
+		Message: err.Error(),
+	})
+	succeeded := trs.GetCondition(apis.ConditionSucceeded)
+	trs.CompletionTime = &succeeded.LastTransitionTime.Inner
+}
+
+// TaskRunStatusFields holds the fields of TaskRun's status.  This is defined
+// separately and inlined so that other types can readily consume these fields
+// via duck typing.
+type TaskRunStatusFields struct {
+	// PodName is the name of the pod responsible for executing this task's steps.
+	PodName string `json:"podName"`
+
+	// StartTime is the time the build is actually started.
+	// +optional
+	StartTime *metav1.Time `json:"startTime,omitempty"`
+
+	// CompletionTime is the time the build completed.
+	// +optional
+	CompletionTime *metav1.Time `json:"completionTime,omitempty"`
+
+	// Steps describes the state of each build step container.
+	// +optional
+	// +listType=atomic
+	Steps []StepState `json:"steps,omitempty"`
+
+	// CloudEvents describe the state of each cloud event requested via a
+	// CloudEventResource.
+	// +optional
+	// +listType=atomic
+	CloudEvents []CloudEventDelivery `json:"cloudEvents,omitempty"`
+
+	// RetriesStatus contains the history of TaskRunStatus in case of a retry in order to keep record of failures.
+	// All TaskRunStatus stored in RetriesStatus will have no date within the RetriesStatus as is redundant.
+	// +optional
+	// +listType=atomic
+	RetriesStatus []TaskRunStatus `json:"retriesStatus,omitempty"`
+
+	// TaskRunResults are the list of results written out by the task's containers
+	// +optional
+	// +listType=atomic
+	TaskRunResults []TaskRunResult `json:"taskResults,omitempty"`
```

Comment:
We're actually going to rename this to just "results" (https://github.com/tektoncd/community/pull/775)

Reply:
It looks `Results` also violate the API naming rule. https://prow.tekton.dev/view/gs/tekton-prow/pr-logs/pull/tektoncd_pipeline/5264/pull-tekton-pipeline-build-tests/1557835200501649408

Reply:
You can rename both the field name and the api field (in the JSON tag) to "results"